test/test_main.cpp

```diff
@@ -141,5 +143,36 @@ TEST(convolution_nhwc, halide) {
 
     ASSERT_LE(norm(ref.reshape(1, 1), dst.reshape(1, 1), NORM_INF), 4e-5f);
 }
+#endif 
+
+// TEST(deconvolution, halide)
+// {
+//     static const int ic = 4;
+//     static const int height = 100;
+//     static const int width = 100;
+//     static const int batch = 72;
+
+//     Mat src({width, height,ic, batch}, CV_32F);
+//     //Mat kernel({4, 4, 4, ic}, CV_32F);
+//     Mat dst({width*2, height*2,ic, batch*2}, CV_32F);
+//     randn(src, 0, 1);
+//     //randn(kernel, 0, 1);
+
+//     voxel_up(src.ptr<float>(), voxel_upscale_const::ker_weight, dst.ptr<float>(),
+//                             ic, width, height, batch);
+//     ASSERT_EQ(true,true);
+// }
+
+TEST(upscale, halide)
+{
+
+    static const int height = 100;
+    static const int width = 100;
+    std::vector<std::string> img_path{"image0001.png"};
```

Comment:
Need to add all images. Even correct deconv cannot manage to process garbage data without necessary number of imread calls:
```cpp
vector<std::string> img_path(72);
for (int i = 1; i <= 72; ++i) {
  img_path = cv::format("image%04d.png", i);
}
   
}